### PR TITLE
Do not report unchanged files in workspace-pull diagnostics

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Small tweak that should ideally save a lot of unnecessary reports being created on the roslyn side and then parsed on the client side.

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs [hunk 1]
```diff
@@ -187,8 +190,14 @@ protected virtual Task WaitForChangesAsync(RequestContext context, CancellationT
                     // Nothing changed between the last request and this one.  Report a (null-diagnostics,
                     // same-result-id) response to the client as that means they should just preserve the current
                     // diagnostics they have for this file.
-                    var previousParams = documentToPreviousDiagnosticParams[diagnosticSource.GetId()];
-                    progress.Report(CreateUnchangedReport(previousParams.TextDocument, previousParams.PreviousResultId));
+                    //
+                    // Note: if this is a workspace request, we can do nothing, as that will be interpreted by the
+                    // client as nothing having been changed for that document.
+                    if (!this.IsWorkspacePullDiagnosticsHandler)
```
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs:196]: Alternative approach - allow CreateUnchangedReport to return a nullable value, and in workspace diagnostics return a null value always.

**Review by dibarbet — APPROVED** [review]

**Comment by dibarbet** [comment]
I think some tests will need updating as well, e.g. TestWorkspaceDiagnosticsWithChangeInNotReferencedProject

File: src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs:196]: fair.  it seemed about the same... i'll noodle on it.

**Comment by CyrusNajmabadi** [comment]
Will set to merge once we confirm that VSCode is ok.

File: src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs [hunk 2]
```diff
@@ -187,8 +190,14 @@ protected virtual Task WaitForChangesAsync(RequestContext context, CancellationT
...
+                    {
+                        var previousParams = documentToPreviousDiagnosticParams[diagnosticSource.GetId()];
+                        progress.Report(CreateUnchangedReport(previousParams.TextDocument, previousParams.PreviousResultId));
```
Comment (kayle) [inline src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs:199]: Do you have the before/after previousParams.TextDocument int versions to compare too? If the version is different, we need to be careful about skipping.

**Review by dibarbet — APPROVED** [review]

**Pull request merged**